### DISCUSSION TITLE
Training using Mpyc

### DISCUSSION THREAD

**Discussion by imtiyazuddin** [lschoe/mpyc#52] [General]
Hi, I have seen examples of inference in the code base, however what changes will I need to do to use it for training (with back propagation etc) , is there any example for it ?

**Comment by lschoe** [comment]
There are some demos involving training namely for ID3 decision trees and for ridge regression. But for neural networks training is not covered yet.

To implement back propagation requires some nontrivial changes. But I guess it can be done using the available support for secure NumPy in MPyC, as used in the `np_cnnmnist.py` demo for example.
